tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -128,3 +128,30 @@ ruleTester.run('no-extraneous-dependencies', rule, {
     }),
   ],
 })
+
+ruleTester.run('no-extraneous-dependencies core-module', rule, {
+  valid: [
+    test({
+      code: 'import "core-module"',
+      settings: { 'import/core-modules': ['core-module'] },
+    }),
+    test({
+      code: 'import "@namespaced/module"',
+      settings: { 'import/core-modules': ['@namespaced/module'] },
+    }),
+    test({
+      code: 'import "@namespaced/module/and/sub/path"',
```

Comment:
Ah, weird. Yeah, #479 is the ideal path to resolving this issue without explicit config (Webpack resolver would report as a `project` or `internal`) but what you've done should theoretically work.

Must be either the `@` or `/` killing it. I suspect the latter...